### PR TITLE
Egl modifiers part 1 (implicit modifiers)

### PR DESCRIPTION
**Pull request by columbarius**
### Description

To properly detect if a client can support DmaBufs it has to announce a seperate EnumFormat with a modifier and one without.
This is a first step announcing the capability for implicit modifiers, a followup MR will add support to query all supported explicit modifiers.

### Motivation and Context

This MR makes dmabuf screensharing compatible with the improved negotiation implemented in
- GNOME: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1939
- KDE: https://invent.kde.org/plasma/kwin/-/merge_requests/1210?diff_id=141932&start_sha=74c2ca0a79a38a598375708e8a11daf285a8c232
- wlroots: https://github.com/emersion/xdg-desktop-portal-wlr/pull/159

### How Has This Been Tested?

This should make obs use dmabuf screensharing with the MR's mentioned above.

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [ ] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

Requires: #5118 
Requires: pipewire 0.3.29

### CODE REVIEW

**Comment by columbarius** [comment]
I don't know how the pipewire source is used on nvidia or other gpus missing support for dmabufs especially in mixed gpu setups. We might want to extend the graphics api to have a check if dmabuf is supported by our current rendering device.

File: plugins/linux-capture/pipewire.c [hunk 1]
```diff
@@ -660,32 +720,43 @@ static void play_pipewire_stream(obs_pipewire_data *obs_pw)
...
+		SPA_VIDEO_FORMAT_BGRA,
+		SPA_VIDEO_FORMAT_RGBA,
+		SPA_VIDEO_FORMAT_BGRx,
+		SPA_VIDEO_FORMAT_RGBx,
+	};
+
+	n_formats = sizeof(formats) / sizeof(formats[0]);
+	n_params = 0;
+
+	params = calloc(2 * n_formats, sizeof(struct spa_param *));
```
Comment (kkartaltepe) [inline plugins/linux-capture/pipewire.c:733]: Please use obs's allocators everywhere. See libobs/util/bmem.h generally just add a `b` in front.

**Comment by columbarius** [comment]
> I don't know how the pipewire source is used on nvidia or other gpus missing support for dmabufs especially in mixed gpu setups. We might want to extend the graphics api to have a check if dmabuf is supported by our current rendering device.

We could solve that issue by using the query functions from part 2 #5121 and check if the extensions are supported. If importing dmabuf is supported without querying modifiers just return N=1 and implicit modifiers, if both are not supported return N=0 and NULL.

**Comment by columbarius** [comment]
This MR is obsoleted will be superseded by #5131 .  There is no sense in either checking if DMA-BUF importing via the required extensions is supported in this plugin, or extending the graphics api with a `_supports_modifiers` function. 

I would like to keep it open to test the DMA-BUF patches in gnome and kde until #5131 is ready for testing.

**Comment by columbarius** [comment]
Since #5131 is now ready for testing and in a much better shape, this MR can be closed.

**Pull request closed without merging**